### ISSUE TITLE
Support for different homeserver url and name

### ISSUE THREAD

**Issue by mtorials** [syphon-org/syphon#163]
I run my own homeserver under a subdomain like in https://matrix.example.com. But the homeserver name is still just example.org. This means my username is also @username:example.org and not @username:matrix.example.com.

__Problem__

I can not log into my server. Using the homeserver URL matrix.example.com as homeserver results in my username then being @username:matrix.example.com.

__Possible Solution__

Without making any UI changes you could check whether the user provides a fully qualified userid or just the username. When the input starts with an @ use the homeserver input as URL and the treat the username input as fully qualified id. Otherwise just use the already existing method of appending the homeserver to the username.

**Comment by ereio** [comment]
I think I've fixed this with the newest version 0.1.2 release after solving this issue: #153

Try logging in now with the username as others see it in the username field, like @username:example.org, and let me know if you're still having trouble.

**Comment by mtorials** [comment]
I tested this now with a few accounts. When I want to sign in on my server the login button loads but nothing happens and I do not get an error message. I also created an account on privacytools.io  to check the behavior with a homeserver I did not configure myself....

With chat.privacytools.io I get an invalid credentials error (username @username:privacytools.io) but I am pretty sure my credentials are right. I also tried loging in on matrix.tedomum.net. This resulted in the same behavior I experienced with my own homerserver: No error message (username: @username:tendomum.net).

I sadly have no idea why I get an error message with privacytools.io and nothing with the other two servers. I also do not how the client knows how to reach the homeserver under a subdomain when only the normal domain is provided. Is there a special query/api endpoint to configure when using synapse with a subdomain?

**Comment by ereio** [comment]
Sorry I haven't gotten back to you sooner, just went through a move.

I just tried signing in with my alternate matrix account 'ereio:privacytools.io' on version 0.1.2 and successfully logged in and pulled all relevant data. 

I think the problem you're experiencing is due to putting the @ symbol before your username. I'll ignore that character on the next release and sorry that slipped through the cracks.

Clients are to resolve the subdomain issue by making a GET call to ```https://homeserver.com/.well-known/matrix/client``` to fetch both the actual homeserver and identity url. Syphon is doing this within version 0.1.2 which is how I could login with the above username.

Let me know if that doesn't work and thanks for replying in depth on the issue. I'm sure others have experienced that too.

**Comment by mtorials** [comment]
Although I was not able to sign in on privacytools.io I was able to sign in on my own homeserver after configuring the `well-known` endpoint. Thank you for your help there!

After logging in the app reported two error: `Failed to fetch current user profile` and `Unknown Error Occured` and no information gets displayed. This seems very similar to the problems you described in #153. But as far as android tells me I am using version 0.1.2. When I tried to relogin the logout button only quickly showed the loading symbol without doing anything. I then reinstalled the app and again logged in with the same result: The described two error and nothing is loading.
